base/range.jl

```diff
@@ -1371,11 +1371,13 @@ in(x::T, r::AbstractRange{T}) where {T} = _in_range(x, r)
 in(x::Integer, r::AbstractUnitRange{<:Integer}) = (first(r) <= x) & (x <= last(r))
 
 in(x::Real, r::AbstractRange{T}) where {T<:Integer} =
-    isinteger(x) && !isempty(r) && x >= minimum(r) && x <= maximum(r) &&
-        (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)
+    isinteger(x) && !isempty(r) &&
+    (iszero(step(r)) ? x == first(r) : (x >= minimum(r) && x <= maximum(r) &&
+    (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)))
```

Comment:
```suggestion
        (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)))
```
(indented as already done for the `AbstractChar` case)